src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -1019,28 +1013,27 @@ var ReactDOMFiberComponent = {
           if (expectedStyle !== serverValue) {
             warnForPropDifference(propKey, serverValue, expectedStyle);
           }
-        } else if (
-          isCustomComponentTag ||
-          DOMProperty.isCustomAttribute(propKey)
-        ) {
-          // $FlowFixMe - Should be inferred as not undefined.
-          extraAttributeNames.delete(propKey);
-          serverValue = DOMPropertyOperations.getValueForAttribute(
-            domElement,
-            propKey,
-            nextProp,
-          );
-          if (nextProp !== serverValue) {
-            warnForPropDifference(propKey, serverValue, nextProp);
+        } else if (DOMProperty.shouldSetAttribute(propKey, nextProp)) {
```

Comment:
Sure. 

**Lines 234-237 (setInitialDOMProperties)**

I replaced all instances of `DOMProperty.properties[propKey] || DOMProperty.isCustomAttribute(propKey)` with a new check: `DOMProperty.shouldSetAttribute`. This provides a similar assertion: should the attribute be written to this DOM element? This function checks ` DOMProperty.properties[propKey]`, but adds additional rules for what are allowed as writeable attributes if property information isn't otherwise specified: 

1. Is the property a reserved word (skip)? 
2. Else is the property an alias of another attribute (skip)?
3. Else is the property null, undefined, or within the properties we pre-approve (keep)?
4. Else is the property value a boolean, number, or string (keep)?

**Lines 278-281 (updateDOMProperties)**

More or less the same thing as `setInitialDOMProperties`. We only want to operate on properties that we know can be written to the element. This is after a series of checks for reserved properties like `dangerouslySetInnerHTML` and `style`. 

**Lines 1022-1024 (diffHydratedProperties)**

I removed this condition because, before this change, only custom elements were allowed to have custom attributes. Now both custom and regular attributes can have custom attributes, so they can share the same logic. This logic follows:

1.*Lines 1019-1026*: If the element is *not* custom, and there is property info (we have a property injection rule for it), get the value for the property, using an aliased attribute name if need be.
2. *Lines 1028-1034*: Otherwise, this is a custom element, or a custom attribute. We know that custom attributes are never assigned as properties, so we can use the same `getValueForAttribute` logic for both (as was the case before this change)